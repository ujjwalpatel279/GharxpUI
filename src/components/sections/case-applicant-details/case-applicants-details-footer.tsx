import { Button } from '../../shared';
import React, { useState, FunctionComponent, ReactElement, MouseEvent } from 'react';
import { CardFooter } from '../../shared/card/card-footer';
import { UploadDocument } from '../upload-document/upload-document';
import { DocumentRefresh } from '../../pages/clients/clients';

import './case-applicants.scss';

interface CardFooterProps {
  clientId?: number;
  documentGridRefresh: DocumentRefresh;
  documentsExist: number;
  handleExpand?: (e: MouseEvent<HTMLButtonElement>) => void;
  expanded: boolean;
}

export const CaseApplicantsDetailsFooter: FunctionComponent<CardFooterProps> = ({
  clientId,
  documentGridRefresh,
  documentsExist,
  expanded,
  handleExpand,
}): ReactElement => {
  const [showAddClient, toggleShowAddClient] = useState(false);
  const [modalVisible, setModalVisible] = useState<boolean>();
  const [customerIdForUpload, setCustomerIdForUpload] = useState<number>(clientId);
  const showModal = () => {
    setModalVisible(true);
    setCustomerIdForUpload(clientId);
  };

  const reloadDocumentGrid = async () => {
    await documentGridRefresh();
  };

  const hideModal = () => {
    setModalVisible(false);
  };
  return (
    <CardFooter>
      <UploadDocument
        customerId={customerIdForUpload}
        reloadDocumentGrid={reloadDocumentGrid}
        show={modalVisible}
        handleClose={hideModal}
      />
      <footer className={`s-card-client__detail ${documentsExist > 3 ? '' : 's-card-client__detail-partial'}`}>
        {documentsExist > 3 && (
          <button onClick={handleExpand} className={'s-card-client__detail__button notes__text u-fs-2'}>
            <b>{expanded ? 'See less' : 'See more'}</b>
          </button>
        )}
        <Button
          type="button"
          variant="secondary"
          icon={showAddClient ? 'close' : 'plus'}
          iconPosition="left"
          onClick={showModal}
          size="small"
        >
          {showAddClient ? 'Close' : 'Add document'}
        </Button>
      </footer>
    </CardFooter>
  );
};
