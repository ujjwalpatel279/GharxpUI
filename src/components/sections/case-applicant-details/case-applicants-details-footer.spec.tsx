import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { CaseApplicantsDetailsFooter } from './case-applicants-details-footer';

const handleRefreshDocuments = jest.fn();
const handleExpanded = jest.fn();
const expanded = true;

describe('shows card document details footer', () => {
  afterEach(cleanup);
  it('show client document card footer', () => {
    const { asFragment } = render(
      <CaseApplicantsDetailsFooter
        documentsExist={2}
        documentGridRefresh={handleRefreshDocuments}
        handleExpand={handleExpanded}
        expanded={expanded}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
