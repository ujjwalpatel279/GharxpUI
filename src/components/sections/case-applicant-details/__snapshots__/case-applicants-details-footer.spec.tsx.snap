// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows card document details footer show client document card footer 1`] = `
<DocumentFragment>
  <div
    class="c-card__footer "
  >
    <div
      class="modal display-none doc-upload"
    >
      <section
        class="modal-main"
      >
        <form>
          <div
            class="c-form-field"
          >
            <label
              class="c-form-field__label"
            >
              File to upload
            </label>
            <input
              accept="image/png, image/jpeg, application/pdf"
              aria-invalid="false"
              id="customerDocument"
              name="customerDocument"
              type="file"
            />
            <p
              class="c-form-field__error"
              role="alert"
            >
              Please select a file of type jpg/png/pdf with a file size less than 4MB
            </p>
          </div>
          <div
            class="c-form-field"
          >
            <label
              class="c-form-field__label"
            >
              Category
            </label>
            <fieldset>
              <legend>
                Select a document category
              </legend>
              <div
                class="l-flex l-flex--wrap"
              />
            </fieldset>
          </div>
          <div
            class="c-form-field"
          >
            <label
              class="c-form-field__label"
            >
              File Description
            </label>
            <input
              aria-invalid="false"
              class="c-input"
              id="fileDescription"
              name="fileDescription"
              placeholder="File Description"
              type="text"
            />
          </div>
          <p
            class="u-mb-1"
          />
          <button
            class="c-btn c-btn--secondary c-btn--small"
            type="button"
          >
            <span>
              Cancel
            </span>
          </button>
          <button
            class="c-btn c-btn--primary c-btn--small disabled upload-button"
            disabled=""
            type="submit"
          >
            <span>
              Upload
            </span>
          </button>
        </form>
      </section>
    </div>
    <footer
      class="s-card-client__detail s-card-client__detail-partial"
    >
      <button
        class="c-btn c-btn--secondary c-btn--small c-btn--icon-left"
        type="button"
      >
        <svg
          class="c-icon"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M7 7v-7h2v7h7v2h-7v7h-2v-7h-7v-2h7z"
          />
        </svg>
        <span>
          Add document
        </span>
      </button>
    </footer>
  </div>
</DocumentFragment>
`;
